File: ruby/src/Behavioral/Chain_Of_Responsibility/DvdSubCategory.rb

from TopTitle import TopTitle

class DvdSubCategory(TopTitle) :

	subCategory = None
	topSubCategoryTitle = None
	parent = None

	def initialize( dvdCategory, subCategory) :
		self.setSubCategory(subCategory)
		self.parent = dvdCategory

	def setSubCategory(self,subCategoryIn) :
		self.subCategory = subCategoryIn

	def getSubCategory(self) : 
		return self.subCategory

	def setCategory(self,categoryIn) :
		self.parent.setCategory(categoryIn)

	def getCategory(self) :
		return self.parent.getCategory()

	def getAllCategories(self) : 
		return '{0}/{1}'.format(self.getCategory(),self.getSubCategory())

	def setTopSubCategoryTitle(self,topSubCategoryTitleIn) :
		self.topSubCategoryTitle = topSubCategoryTitleIn

	def getTopSubCategoryTitle(self) :
		return self.topSubCategoryTitle

	def setTopCategoryTitle(topCategoryTitleIn) : 
		self.parent.setTopCategoryTitle(topCategoryTitleIn)

	def getTopCategoryTitle(self) : 
		return self.parent.getTopCategoryTitle()

	def getTopTitle(self) :
		return self.getTopSubCategoryTitle() if self.getTopSubCategoryTitle() else self.parent.getTopTitle()

File: java/src/Behavioral/Chain_Of_Responsibility/DvdSubCategory.java
package Behavioral.Chain_Of_Responsibility;


//DvdSubCategory.java - the Middle Class in the Chain

public class DvdSubCategory implements TopTitle {  
   private String subCategory; 
   private String topSubCategoryTitle;
   private DvdCategory parent;
   
   public DvdSubCategory(DvdCategory dvdCategory, String subCategory) {
        this.setSubCategory(subCategory); 
        this.parent = dvdCategory;
   }    
   
   public void setSubCategory(String subCategoryIn) {
       this.subCategory = subCategoryIn;
   }
   public String getSubCategory() {
       return this.subCategory;
   }
   public void setCategory(String categoryIn) {
       parent.setCategory(categoryIn);
   }
   public String getCategory() {
       return parent.getCategory();
   }
   public String getAllCategories() {
       return (getCategory() + "/" + getSubCategory());
   }
   
   public void setTopSubCategoryTitle(String topSubCategoryTitleIn) {
       this.topSubCategoryTitle = topSubCategoryTitleIn;
   }
   public String getTopSubCategoryTitle() {
       return this.topSubCategoryTitle;
   }
   public void setTopCategoryTitle(String topCategoryTitleIn) {
       parent.setTopCategoryTitle(topCategoryTitleIn);
   }
   public String getTopCategoryTitle() {
       return parent.getTopCategoryTitle();
   }
   
   public String getTopTitle() {
       if (null != getTopSubCategoryTitle()) {
           return this.getTopSubCategoryTitle();
       } else {
           //System.out.println("no top title in Category/SubCategory " + getAllCategories());
           return parent.getTopTitle();
       }
   }
}
